frontend/packages/ux-editor/src/components/EditFormComponent.tsx

```diff
@@ -0,0 +1,203 @@
+    if (JSON.stringify(component) !== JSON.stringify(props.component)) {
+      handleSaveChange(component);
+      if (props.id !== component.id) {
+        dispatch(
+          FormLayoutActions.updateFormComponentId({
+            newId: component.id,
+            currentId: props.id,
+            org,
+            app,
+          })
+        );
+      }
+    }
+  };
+
+  const handleDiscard = (): void => {
+    setComponent({ ...props.component, id: props.id });
+    setMode(EditFormComponentMode.Closed);
+  };
+
+  const handleSaveChange = (callbackComponent: FormComponentType) =>
+    updateFormComponent({
+      id: props.id,
+      updatedComponent: callbackComponent,
+    });
+
+  return (
+    <div className={cn(classes.wrapper, isPreviewMode && classes.previewMode)} role='listitem'>
+      <div className={classes.formComponentWithHandle}>
+        <div ref={props.dragHandleRef} className={classes.dragHandle}>
+          <DragHandle />
+        </div>
+        <div
+          className={classes.formComponent}
+          tabIndex={0}
+        >
+          {isPreviewMode && component && (
+            <ComponentPreview
+              component={component}
+              handleComponentChange={handleComponentChangeAndSave}
+              layoutName={selectedLayout}
+            />
+          )}
+          {isEditMode && component && (
+            <EditModalContent
+              component={deepCopy(component)}
+              handleComponentUpdate={handleComponentUpdate}
+            />
+          )}
+          {(mode === EditFormComponentMode.Closed || !component) && (
+            <div className={classes.formComponentTitle}>
+              <i className={componentIcons[component.type] || 'fa fa-help-circle'} />
+              {component.textResourceBindings?.title
+                ? truncate(getTextResource(component.textResourceBindings.title, textResources), 80)
+                : getComponentTitleByComponentType(component.type, t) ||
+                  t('ux_editor.component_unknown')}
+            </div>
+          )}
+        </div>
+      </div>
+      {!isEditMode && (
+        <div className={classes.buttons}>
+          <Button
+            data-testid='component-delete-button'
+            color={ButtonColor.Secondary}
+            icon={<TrashIcon />}
+            onClick={handleComponentDelete}
+            tabIndex={0}
+            title={t('general.delete')}
+            variant={ButtonVariant.Quiet}
+          />
+          <Button
+            color={ButtonColor.Secondary}
+            icon={<PencilIcon />}
+            onClick={handleOpenEdit}
+            tabIndex={0}
+            title={t('general.edit')}
+            variant={ButtonVariant.Quiet}
+          />
+          {isPreviewable && (
+            <Button
+              color={ButtonColor.Secondary}
+              icon={<MonitorIcon title={t('general.preview')} />}
+              onClick={() => setMode(EditFormComponentMode.Preview)}
```

Comment:
Not really part of this change, as this is "as-is" atm, but I've noticed that this button doesn't make it possible to disable preview mode again after it's enabled. Could you set up a "handlePreviewMode"-type function that allows user to toggle preview-mode on/off while you are in here anyways? 😇

Reply:
That was easy to fix so I did it in this PR. I fixed it by replacing the preview button with a close button in preview mode. Hope that's ok.
And while I was testing it I discovered a bug with the drag&drop that I didn't see previously 😕 I pushed a commit that will fix it.

Reply:
I did some more testing while working on the EditFormComponent and it actually looks a bit weird to display a close button in preview mode too. So I just did a toggle button using the same preview icon as you suggested.